File: src/services/billetera_service.ts
import { api } from './api';

export const billeteraServicio = {
  miBilletera: () => api.get('/billetera').then((r) => r.data),
  generarQr: () => api.get('/billetera/qr').then((r) => r.data),
  pagar: (lineaId: string) => api.post('/billetera/pagar', { lineaId }).then((r) => r.data),
  stripeCheckout: (monto: number) => api.post('/billetera/stripe/checkout', { monto }).then((r) => r.data),
  stripeConfirmar: (sessionId: string) => api.post('/billetera/stripe/confirmar', { sessionId }).then((r) => r.data),
  miHistorial: () => api.get('/billetera/historial').then((r) => r.data),
};

File: lib/services/billetera_service.dart
import 'package:dio/dio.dart';

class BilleteraService {
  final Dio dio;
  BilleteraService(this.dio);

  Future<Map<String, dynamic>> miBilletera() async {
    final r = await dio.get('/billetera');
    return r.data;
  }

  Future<Map<String, dynamic>> generarQr() async {
    final r = await dio.get('/billetera/qr');
    return r.data;
  }

  Future<Map<String, dynamic>> pagar(String lineaId) async {
    final r = await dio.post('/billetera/pagar', data: {'lineaId': lineaId});
    return r.data;
  }

  Future<Map<String, dynamic>> stripeCheckout(double monto) async {
    final r = await dio.post('/billetera/stripe/checkout', data: {'monto': monto});
    return r.data;
  }

  Future<Map<String, dynamic>> stripeConfirmar(String sessionId) async {
    final r = await dio.post('/billetera/stripe/confirmar', data: {'sessionId': sessionId});
    return r.data;
  }

  Future<Map<String, dynamic>> miHistorial() async {
    final r = await dio.get('/billetera/historial');
    return r.data;
  }
}
